### PR TITLE
Update Martha release process [BT-607]

### PR DESCRIPTION
**Pull request by jgainerdewar**
This is done as part of an effort to standardize the release process for non-monolith products.

### CODE REVIEW

File: docs/release-process.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+Martha Release Process
+=========
+
+Martha's release process is documented and tracked through a Jira checklist. 
+
+Martha is updated rarely enough that we typically include a release in the 
+work of each ticket. Navigate to the ticket associated with the work you're 
+releasing, or create a new ticket for the release. In the Jira ticket, click 
+on Add Checklist, then click on the button with three dots in the upper right
```
Comment (breilly2) [inline docs/release-process.md]: "Add Checklist" is this button, right?
Screen Shot 2022-04-14 at 3 07 52 PM

**Review by breilly2 — APPROVED** [review]
👍  Assuming I correctly identified the "Add Checklist" button. It might be worth adding more description about how to find this button; it wasn't immediately obvious to me.

File: docs/release-process.md (hunk 1 above)
Comment (jgainerdewar) [reply docs/release-process.md]: Yes! In other views of the ticket there's text in it. I'll expand the description here.

**Review by mspector — APPROVED** [review]
Seems clear to me!

**Pull request merged**